src/java/main/org/apache/zookeeper/server/ServerStats.java

```diff
@@ -54,9 +57,10 @@ synchronized public long getMinLatency() {
         return minLatency == Long.MAX_VALUE ? 0 : minLatency;
     }
 
-    synchronized public long getAvgLatency() {
+    synchronized public double getAvgLatency() {
         if (count != 0) {
-            return totalLatency / count;
+            String avgLatency = df.format((double)totalLatency / count);
+            return Double.parseDouble(avgLatency);
```

Comment:
Why don't you this way ?
return totalLatency *1.0 / count

Using format/parse is costly and it can lead to unexpected results, expecially with floating point values

Reply:
@eolivelli totalLatency *1.0 / count will get e.g 0.7649024988323213 (too long),but I want to that the results tally up to four decimal places e.g 0.7649

Reply:
The conversion you are doing does not force a given number of digits

You should perform the division and the return a string fornatted with a max number of decimal digits.

Or you can calculate the value as integer with some trick
Like
return ( totalLatency *100 / count) / 100.0